core/src/main/java/com/datastax/oss/driver/api/core/metadata/diagnostic/Status.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.metadata.diagnostic;
+
+/** The status of a {@link Diagnostic}. */
+public enum Status {
+
+  // the order of enums matter, see mergeWith
```

Comment:
If we're going to go down that path, though, isn't precedence a context-sensitive operation?  Perhaps the ordering shouldn't be built into the type at all but should be implemented as a Comparator which would allow different callers to define whatever prioritization makes sense to them.  At the moment there's a single consumer of this functionality so moving that logic into a Comparator at the current call site seems pretty straightforward.

Reply:
Changed to make the order irrelevant.